### PR TITLE
Fix Pyccel checks

### PR DESCRIPTION
**Pull request by EmilyBourne**
The PR test which checks that all AST nodes are following the expected conventions (`__slots__` and `_attribute_nodes` defined) is not running on subfolders. As new PRs are introducing subfolders this needs fixing ASAP. This PR fixes that

### CODE REVIEW

**Comment by EmilyBourne** [comment]
/bot run coverage macosx windows pylint docs

**Review by Farouk-Echaref — APPROVED** [review]
Looks good to me

File: ci_tools/check_slots.py [hunk 1]
```diff
@@ -92,8 +92,8 @@ def fill_dictionary(title, message, file, start_line, end_line, annotation_level
     args = parser.parse_args()
 
     # Get ast modules
-    ast_folder = os.path.dirname(ast.__file__)
-    ast_modules = [mod[:-3] for mod in os.listdir(ast_folder) if mod != '__init__.py' and mod.endswith('.py')]
+    ast_folder = pathlib.Path(ast.__file__).parent
+    ast_modules = ['.'.join(f.relative_to(ast_folder).parts)[:-3] for f in ast_folder.rglob('*.py') if f.parts[-1] != '__init__.py']
```
Comment (yguclu) [inline ci_tools/check_slots.py:96]: Why are you taking the last three characters of a string? Should it not be this instead?
```suggestion
    ast_modules = ['.'.join(f.relative_to(ast_folder).parts[:-3]) for f in ast_folder.rglob('*.py') if f.parts[-1] != '__init__.py']
```

Comment (EmilyBourne) [reply ci_tools/check_slots.py:96]: I'm not taking the last three characters. I am stripping the last three characters. In other words removing the extension `.py` so I can create a module name that can be imported

Comment (yguclu) [reply ci_tools/check_slots.py:96]: True! I read it in the opposite way! 😅

**Review by yguclu — APPROVED** [review]

**Pull request merged**